### ISSUE TITLE
Customize each block individually?

### ISSUE THREAD

**Issue by Lockoloop** [woocommerce/woocommerce-gutenberg-products-block#7175]
Hi, I would like to be able to customize the individual blocks a bit more.
I managed to customize most of the visuals enqueuing my own blocks css sheet (as it looked wrong with my home-made theme), but for instance I would like to be able to customize the "proceed to checkout" page (as I'm using an icon instead, as you can see on my sidecart).
I wish I could also add relevant information for my shop such as the weight of my products and so on.
I don't know much about customizing blocks, do they work like templates with filters and do/add actions?

Thank you.

https://www.lockoloop.nl/devstage/shop/

**Comment by danielwrobert** [comment]
Hi @Lockoloop! Thanks for reaching out.

You can customize styling for blocks via the `register_block_style` function (and `unregister_block_style`, if needed). You can read a bit more about this approach in the following articles:

- https://fullsiteediting.com/lessons/custom-block-styles/
- https://awhitepixel.com/blog/wordpress-gutenberg-custom-block-styles/
- https://themeshaper.com/2019/02/15/customizing-gutenberg-blocks-with-block-styles/

It sounds like you may have already done this?

If you need to go further than that, it is possible to extend an existing block using [block filters](https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/). You can see a couple of examples in the following articles:

- https://jeffreycarandang.com/extending-gutenberg-core-blocks-with-custom-attributes-and-controls/
- https://www.liip.ch/en/blog/how-to-extend-existing-gutenberg-blocks-in-wordpress

> for instance I would like to be able to customize the "proceed to checkout" page

For this, it sounds like you may want to override the WooCommerce template that lives at `woocommerce/templates/cart/proceed-to-checkout-button.php`. You can read more about doing so here: https://woocommerce.com/document/template-structure/.

I hope that answers your questions!

I'm going to go ahead and close this out for now. If you have any additional questions, feel free to respond and we can re-open the Issue, if necessary.

Additionally, you can post in the plugin support forum for any general questions/concerns: https://wordpress.org/support/plugin/woo-gutenberg-products-block/.

**Comment by Lockoloop** [comment]
Hi Daniel,
Thanks for the exhaustive reply.

As you guessed, I already started to overwrite the blocks CSS (by dequeuing/enqueuing).

But about the proceed to checkout button, I realized that I made a typo, I meant that I want to customize the "proceed to checkout" **block**, not the page: I am already replacing the checkout link with a button (through the functions file), but it seems like it doesn't apply to the block:
```
//Replaces To checkout button
function woocommerce_button_proceed_to_checkout() { ?>
 <a href="<?php echo esc_url( wc_get_checkout_url() ); ?>"> 
 <?php esc_html_e( '', 'woocommerce' ); ?><button class="LCKL_button"><i class="icon-checkout"></i></button>
 </a>
 <?php
}
```
Are they independent?
And should I do that using the block filter? To at least make the text editable? (so I can remove it and insert my icon with a content in a :before class)
Or did I miss an option somewhere?

Thank you!
